### PR TITLE
Refactor SQLite handler

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

It might be easier to review a result than changes:
* https://github.com/AlekSi/FerretDB/tree/metadata/internal/backends
* https://github.com/AlekSi/FerretDB/tree/metadata/internal/handlers/sqlite

## Readiness checklist

* [x] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [x] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [ ] I updated user documentation.
* [x] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/backends/sqlite/collection.go [hunk 1]
```diff
@@ -32,304 +30,146 @@ import (
...
-		if err != nil {
-			// TODO: check error
-			tx.Rollback()
-		}
-	}()
+	// TODO https://github.com/FerretDB/FerretDB/issues/2750
 
-	var inserted int64
+	db := c.r.DatabaseGetExisting(ctx, c.dbName)
+	query := fmt.Sprintf(`INSERT INTO %q (_ferretdb_sjson) VALUES (?)`, c.r.CollectionToTable(c.name))
```
Comment (chilagrow) [inline internal/backends/sqlite/collection.go:79]: We are changing the column name here, I believe sqlite is still experimental so backwards compatibility isn't an issue I suppose.

File: internal/backends/sqlite/collection.go [hunk 2]
```diff
@@ -32,304 +30,146 @@ import (
...
 	}
 
-	var committed bool
+	tableName := c.r.CollectionToTable(c.name)
 
-	defer func() {
-		if committed {
-			return
-		}
+	query := fmt.Sprintf(`UPDATE %q SET _ferretdb_sjson = ? WHERE json_extract(_ferretdb_sjson, '$._id') = ?`, tableName)
```
Comment (chilagrow) [inline internal/backends/sqlite/collection.go:122]: This is great :hugs: , only tableName now needs `%q`

File: internal/handlers/sqlite/msg_insert.go [hunk 3]
```diff
@@ -41,28 +40,31 @@ func (h *Handler) MsgInsert(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
...
-	if len(res.Errors) > 0 {
-		var errs *commonerrors.WriteErrors
-
-		for i := 0; i < len(res.Errors); i++ {
-			errs.Append(err, int32(i))
-		}
-
-		replyDoc = errs.Document()
-	}
+	// TODO
```
Comment (chilagrow) [inline internal/handlers/sqlite/msg_insert.go:58]: I suppose this is okay like this for now?

**Review by chilagrow — APPROVED** [review]
Looks good to me, just a couple of question

File: internal/backends/sqlite/collection.go (hunk 1 above)
Comment (AlekSi) [reply internal/backends/sqlite/collection.go:79]: Yeah, and now we use a reserved prefix so we could implement #226 more easily

File: internal/handlers/sqlite/msg_insert.go (hunk 3 above)
Comment (AlekSi) [reply internal/handlers/sqlite/msg_insert.go]: Added the issue link

File: internal/backends/sqlite/collection.go (hunk 1 above)
Comment (chilagrow) [reply internal/backends/sqlite/collection.go:79]: Yeah that's good direction :+1:

File: internal/backends/sqlite/collection.go (hunk 2 above)
Comment (chilagrow) [reply internal/backends/sqlite/collection.go:122]: That's convenient for this!

**Review by chilagrow — APPROVED** [review]
:rocket:

**Review by rumyantseva — APPROVED** [review]
I mostly checked structure and what is in which package. The approach looks reasonable!

I need an explanation about packages inside `sqlite` handler: why `pool` is inside `metadata` and what is considered as `metadata` in geneal.

**Comment by AlekSi** [comment]
That's exactly the problem I mentioned on DS calls. Basically, `metadata` and `pool` could be in the same package – except that they will be able to access each other's unexported methods and fields, and that would be a mess.

**Pull request merged**